draft-looker-cfrg-bbs-signatures.md

```diff
@@ -677,7 +682,7 @@ Procedure:
 5.     n = n + 1
 6.     generator_i = Identity_G1
 7.     candidate = hash_to_curve_g1(v, generator_dst)
-8.     if candidate in (P1, generator_1, ..., generator_i):
+8.     if candidate in (generator_1, ..., generator_i):
```

Comment:
@BasileiosKal @OR13  most likely a dumb question(s) from me, as I'm not an expert in BBS+:
1) Is it OK to have a collision for some generator_i == P1?
2) Or is it guaranteed that hash_to_curve_g1 <> P1? (why?)

may be it's worth a note here.

Reply:
cc @mikelodder7 @tplooker

Reply:
That’s actually a great question😄

## TL;TR
It's not OK to have a collision, but the probability of having a collision with P1 is practically 0.

## In detail
That probability is ~2^-128. This comes from the collision resistance of hash_to_curve_g1 (which comes from the collision resistance of the underlying hash function+the security level of the curve). This guarantees that unless you create more than 2^128 generators, generator_i will be different from P1 (note that they are created from different inputs to `hash_to_curve_g1`).

If that security level is deemed not enough in the future, there are alternative curves and hash functions that can be used, with higher security levels.

We have checks like that throughout the document, but we are currently debating if we should keep all of them. The tradeoff is that they guard against events that could compromise security, however, the probability of those events is practically 0. Also, we are worried that the more complicated draft could lead to implementation bugs like infinite loops (since we keep trying in a loop until we get a valid result).

A note is a good idea. We do have an [issue](https://github.com/decentralized-identity/bbs-signature/issues/206) open, to debate on what the best way to describe those requirements would be. No matter the case though, we will soon add text that describes all those considerations in detail.

Reply:
Agreed. The likelihood of a collision is so low it’s not worth checking. Just a note about it is good enough

Reply:
Thanks for the details, that makes perfect sense. My understanding was similar, but the next things is actually the source for my confusion: 
What about of the collusion for `generator_i` and `generator_j`? isn't it the same to `P1` vs `generator_i` ?

Should we remove the whole check in this line?

Reply:
Collisions between `generator_i` and `generator_j` are a bit different. Because we are essentially checking for (any) duplicates (rather than equality with a fixed value, i.e., `generator_i = P1`), we must also consider the [birthday paradox](https://en.wikipedia.org/wiki/Birthday_problem) when calculating the probability of that event.
 
That said, it probably won’t matter enough in practice, and the probability will still be close to ~2^-128. So most likely we will remove that check too. It’s a bit more complicated case though so we want to be 100% sure first.

Reply:
ah yes, I've missed the effect of the birthday paradox. Not sure how much actual effect it will have for avg number of msgs per credential for such small probability per single collision, but agree: let's keep the rest of the check for now.
Thanks for the explanation.